tests/robottelo/test_datafactory.py

```diff
@@ -0,0 +1,118 @@
+"""Tests for module ``robottelo.helpers``."""
+import unittest2
+
+from robottelo.config import settings
+from robottelo.datafactory import (
+    generate_strings_list,
+    invalid_id_list,
+    invalid_names_list,
+    invalid_values_list,
+    InvalidArgumentError,
+    valid_data_list,
+    valid_labels_list,
+    valid_names_list,
+)
+
+
+class SmokeTestCase(unittest2.TestCase):
+    """Tests for smoke decorator"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Backup the config smoke property"""
+        cls.smoke_backup = settings.smoke
+
+    def test_smoke_false(self):
+        """Tests if smoke = 0 returns all data points"""
+        settings.smoke = False
+        self.assertEqual(len(generate_strings_list()), 7)
+        self.assertEqual(len(invalid_id_list()), 4)
+        self.assertEqual(len(invalid_names_list()), 7)
+        self.assertEqual(len(invalid_values_list()), 10)
+        self.assertEqual(len(valid_labels_list()), 2)
+        self.assertEqual(len(valid_data_list()), 7)
+        self.assertEqual(len(valid_names_list()), 15)
+
+    def test_smoke_true(self):
+        """Tests if smoke = 0 returns only *one* data point"""
+        settings.smoke = True
+        self.assertEqual(len(generate_strings_list()), 1)
+        self.assertEqual(len(invalid_id_list()), 1)
+        self.assertEqual(len(invalid_names_list()), 1)
+        self.assertEqual(len(invalid_values_list()), 1)
+        self.assertEqual(len(valid_data_list()), 1)
+        self.assertEqual(len(valid_labels_list()), 1)
+        self.assertEqual(len(valid_names_list()), 1)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Reset the config smoke property"""
+        settings.smoke = cls.smoke_backup
+
+
+class GenerateStringListTestCase(unittest2.TestCase):
```

Comment:
Join all return type checking tests into a common test case instance.

Chain the data factories like:

```python
for string in itertools.chain(generate_strings_list(), invalid_names_list(), ...):
    self.assertIsInstance(string, unicode)
```